src/adam.cpp

```diff
@@ -89,19 +89,13 @@ miopenStatus_t Adam(Handle& handle,
                                                   foundInfDesc,
                                                   stepInDesc,
                                                   stepOutDesc,
-                                                  step,
-                                                  lr,
-                                                  beta1,
-                                                  beta2,
-                                                  weight_decay,
-                                                  eps,
                                                   amsgrad,
-                                                  maximize,
+                                                  true,
                                                   adamw,
                                                   is_amp};
 
     const auto invoke_params = [&]() {
-        auto tmp = adam::InvokeParams{};
+        auto tmp = adam::AdamInvokeParams{};
```

Comment:
[Q] Why this rename is necessary? `adam::InvokeParams` looks good to me.

UPDATE: Now I see. Let's use the same InvokeParams for Adam and AdamW. You can use union to minimize footprint of InvokeParams, if you wish (up to you).